src/main/scala/xerial/sbt/Sonatype.scala

```diff
@@ -76,6 +85,26 @@ object Sonatype extends AutoPlugin {
     )
   )
 
+  sealed trait ProjectHosting {
+    def homepage: String
+    def scmUrl: String
+    def developer: Developer
+    def scmInfo                     = ScmInfo(url(homepage), scmUrl)
+    def developers: List[Developer] = List(developer)
+  }
+
+  abstract class BaseHosting(domain: String) extends ProjectHosting {
```

Comment:
I'm still wondering whether this base class structure is good or not because:
- It has duplicate developer and developers fields, which cannot be override in GitHubHosting
- Almost all field parameters are shared between GitHubHosting and GitLabHosting.

I'd rather define a factory method like:
```scala
def gitHubHosting(user:String, repositoryName:String, email:String) = Some(Hosting(domain = "https://github/...", ..))
def gitHubHosting(user:String, repositoryName:String, email:String) = Some(...)
```